### PR TITLE
Fixes #5838

### LINKED ISSUE #5838: Additional output is incorrect when FP count simulation is active
Simple reproducible:
```python
from rdkit import Chem
from rdkit.Chem import rdFingerprintGenerator

smiles = 'OCCCCCCN'
mol = Chem.MolFromSmiles(smiles)

generator = rdFingerprintGenerator.GetTopologicalTorsionGenerator(fpSize=2048, torsionAtomCount=6)
additional_output = rdFingerprintGenerator.AdditionalOutput()
additional_output.AllocateBitPaths()
fingerprint = generator.GetFingerprint(mol, additionalOutput=additional_output)
bit_paths = additional_output.GetBitPaths()
print('bits', [bit for bit in range(2048) if fingerprint[bit] > 0])
print('additional_output', list(bit_paths.keys()))

bits [1576, 1716, 1960]
additional_output [394, 429, 490]
```
The bits are 2048-based in the returned fingerprint, but only 512-based in the additional output, which is inconsistent.

### PR DESCRIPTION
**Pull request by greglandrum**
Adds logic to copy the additional info from the non-count-simulation bits over into the corresponding slots for the count-simulation bits.

### CODE REVIEW

**Comment by greglandrum** [comment]
Looks like I forgot to check something in here... I will figure this out.

File: Code/GraphMol/Fingerprints/FingerprintGenerator.cpp [hunk 1]
```diff
@@ -314,10 +389,32 @@ FingerprintGenerator<OutputType>::getFingerprint(
     const ROMol &mol, FingerprintFuncArguments &args) const {
   std::uint32_t effectiveSize = dp_fingerprintArguments->d_fpSize;
   if (dp_fingerprintArguments->df_countSimulation) {
-    // effective size needs to be smaller than result size to compansate for
+    // effective size needs to be smaller than result size to compensate for
     // count simulation
     effectiveSize /= dp_fingerprintArguments->d_countBounds.size();
   }
+
+  AdditionalOutput countSimulationOutput;
```
Comment (bp-kelley) [inline Code/GraphMol/Fingerprints/FingerprintGenerator.cpp:397]: Not a big deal, but we could make this code not duplicated in the two functions.

File: Code/GraphMol/Fingerprints/catch_tests.cpp [hunk 2]
```diff
@@ -549,9 +549,9 @@ TEST_CASE("AtomPairGenerator bit info", "[fpgenerator][AP]") {
   SECTION("folded bitInfo") {
     {
       AdditionalOutput::bitInfoMapType expected = {
-          {351, {{0, 1}}},
```
Comment (bp-kelley) [inline Code/GraphMol/Fingerprints/catch_tests.cpp:552]: It appears that this changes the output, do we need a release note?

Comment (greglandrum) [reply Code/GraphMol/Fingerprints/catch_tests.cpp:552]: The title of the bug will be in the release notes, so I just made it a bit more descriptive.

File: Code/GraphMol/Fingerprints/FingerprintGenerator.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/Fingerprints/FingerprintGenerator.cpp:404]: Yep, good idea. I did that.

**Comment by greglandrum** [comment]
@ptosco : will you have time to take a look at this in the next couple of days or should I move the issue to the 2022.09.5 release?

**Comment by ptosco** [comment]
@greglandrum Yes, I’ll review it this weekend.

**Review by ptosco — APPROVED** [review]
LGTM. Thanks a lot for fixing this!

**Pull request merged**